### PR TITLE
Pipeline filenames/directory structuring.

### PR DESCRIPTION
**Pull request by mitchdenny**
Based on a whiteboard conversation we had I've renamed the PR pipeline definitions to ci.yml, and also moved the templates etc under the eng/ folder. Once this is merged I'll rename the pipeline definitions to match our agreed convention.

### CODE REVIEW

File: eng/pipelines/ci.yml [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
-  branches:
-    include:
-    - users/mitchdenny/pr-pipelines
   paths:
     include: # Changes in these directories cause us to build everything for validation purpoes.
     - eng/
     - .azure-pipelines/
 
+trigger:
+- naster
```
Comment (weshaggard) [inline eng/pipelines/ci.yml]: LOL I think you want master

File: eng/pipelines/ci.yml [hunk 2]
```diff
@@ -2,16 +2,16 @@
 # This file is generated automatically and any changes will be lost.
```
Comment (weshaggard) [inline eng/pipelines/ci.yml]: didn't we want to call this file client.yml, at least for now?

File: sdk/eventhub/ci.yml [hunk 3]
```diff
@@ -10,4 +10,4 @@ variables:
   ProjectFile: sdk/eventhub/Microsoft.Azure.EventHubs.sln
 
 jobs:
-- template: ../../.azure-pipelines/templates/jobs/archetype-sdk-client.yml
\ No newline at end of file
+- template: templates/jobs/archetype-sdk-client.yml
```
Comment (weshaggard) [inline sdk/eventhub/ci.yml]: How does it resolve this relative path?

Comment (mitchdenny) [reply sdk/eventhub/ci.yml]: Whoops - fixed this. Makes me wonder whether I should introduce a global which is $(Eng.Path) etc.

File: eng/pipelines/ci.yml (hunk 2 above)
Comment (mitchdenny) [reply eng/pipelines/ci.yml]: Fixed

File: eng/pipelines/ci.yml (hunk 1 above)
Comment (mitchdenny) [reply eng/pipelines/ci.yml]: naster is the new master. Fixed.

File: eng/pipelines/client.yml [hunk 4]
```diff
@@ -2,16 +2,16 @@
...
 pr:
-  branches:
-    include:
-    - users/mitchdenny/pr-pipelines
   paths:
     include: # Changes in these directories cause us to build everything for validation purpoes.
     - eng/
     - .azure-pipelines/
 
+trigger:
```
Comment (weshaggard) [inline eng/pipelines/client.yml]: Did you ever figure out whether or not master is the default trigger or if it is all?

**Comment by mitchdenny** [comment]
Looks like Jenkins got stuck finalizing the build - it completed successfully though.

**Comment by mitchdenny** [comment]
After looking through the docs, I think things are ambiguous if we specify path filters and not branch filters so I decided to put a branch filter in for the PR trigger on master. We may remove it again but I want to make sure that when this PR lands we are good to go.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**